core/src/main/web/app/mainapp/mainapp.js

```diff
@@ -355,6 +356,16 @@
                     notebookUri, uriType, readOnly, format, notebookModel, edited, sessionId, true);
               });
             },
+            fromImport: function(sessionId) {
+              var notebook = $sessionStorage.importedNotebook;
+              var notebookUri = null;
+              var uriType = null;
+              var readOnly = true;
+              var format = null;
+              var notebookModel = bkNotebookVersionManager.open(notebook);
+              loadNotebookModelAndResetSession(
+                  notebookUri, uriType, readOnly, format, notebookModel, false, sessionId, false);
```

Comment:
I'm just loading the notebook here right from the JSON received from the server which is resulting in inputs as arrays, yielding empty input fields.